codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/waiters/RustJmespathShapeTraversalGenerator.kt

```diff
@@ -790,7 +805,13 @@ private fun RustType.isNumber(): Boolean = this.dereference().let { it is RustTy
 
 private fun RustType.isBool(): Boolean = this.dereference() is RustType.Bool
 
-private fun RustType.isDoubleReference(): Boolean = this is RustType.Reference && this.member is RustType.Reference
+private fun RustType.isDoubleReference(): Boolean = this.refCount() > 1
+
+private fun RustType.refCount(): Int =
```

Comment:
I can probably revert this change. I was going to use `refCount`, but then didn't end up going down that path.